proposals/2020-07-01_aws-kms-keyring-redesign/change.md

```diff
@@ -0,0 +1,327 @@
+   authorization check for the encrypted data key.
+2. The keyring is unable to write a decryption contract,
+   but it MAY fulfill any AWS KMS decryption contract for the configured AWS region and, if provided, the AWS account ID.
+   1. Fulfilling any AWS KMS decryption contract MUST be sufficient to succeed.
+
+**Initialization**
+
+On keyring initialization, an AWS KMS symmetric region discovery keyring MUST define the following:
+
+- AWS SDK client (and the AWS SDK client’s region)
+
+On keyring initialization, an AWS KMS symmetric region discovery keyring MAY define the following:
+
+- AWS account ID
+- Grant Tokens
+
+### Keyring-Producing Operations
+
+**Please note the following names are not finalized.**
+
+The ESDK MUST provide customers a way to use the following:
+
+1. **AWS KMS symmetric multi-CMK keyring**
+   1. MUST be configured with a list of key names
+   2. MUST be configured with a key name identifying the [generator](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#generator)
+   3. MAY be configured with a list of string [grant tokens](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#grant-tokens)
+      to be included in all AWS KMS calls
+   4. MUST configure a multi-keyring of AWS KMS symmetric keyrings...
+      1. with a single AWS KMS symmetric keyring for each configured key name
+      2. with each AWS KMS symmetric keyring configured with the configured grant tokens, if applicable
+      3. where the order of the child keyrings matches the order of the configured key names
+2. **AWS KMS symmetric multi-region discovery keyring**
+   1. MUST be configured with a list of AWS regions
+   2. MAY be configured with an AWS account ID
+   3. MAY be configured with a list of string [grant tokens](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#grant-tokens)
+      to be included in all AWS KMS calls
+   4. MUST configure a multi-keyring of AWS KMS symmetric region discovery keyrings...
+      1. with a single AWS KMS symmetric region discovery keyring for each configured AWS region
+      2. with each AWS KMS symmetric region discovery keyring configured with the configured AWS account ID, if applicable
+      3. with each AWS KMS symmetric region discovery keyring configured with the configured grant tokens, if applicable
+      4. where the order of the child keyrings matches the order of the configured AWS regions
+
+**All keyring-producing operations MUST...**
+
+1. Initialize/configure the AWS SDK client(s) required for communicating with AWS KMS for the given key names/AWS regions
+2. Configure each AWS KMS keyring with the AWS SDK client it needs to communicate with AWS KMS
+3. Be configurable with an optional AWS SDK client configuration (client config)
```

Comment:
Can you link to an example here?

Reply:
Added...

   (examples include: [Java ClientConfiguration](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/section-client-configuration.html), [Python botocore.config](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html), etc.)